org.idempiere.test/src/org/idempiere/test/base/DocumentStatusTest.java

```diff
@@ -0,0 +1,165 @@
+import org.compiere.model.MOrder;
+import org.compiere.model.X_PA_DocumentStatusAccess;
+import org.compiere.util.DB;
+import org.compiere.util.Env;
+import org.idempiere.test.AbstractTestCase;
+import org.junit.jupiter.api.Test;
+/**
+ * Unit test cases for document status indicator
+ * @author Deepak Pansheriya
+ *
+ */
+public class DocumentStatusTest extends AbstractTestCase {
+
+	public DocumentStatusTest() {
+	}
+
+	private static final int USER_GARDENADMIN = 101;
+	private static final int USER_GARDENUSER = 102;
+	private static final int USER_SUPERUSER = 100;
+
+	private static final int ROLE_GARDENWORLDADMIN = 102;
+	private static final int ROLE_GARDENWORLDUSER = 103;
+	private static final int ROLE_GARDENWORLD_ADMIN_NOT_ADVANCED = 200001;
+
+	private static final int WINDOW_ID_INVOICE_CUSTOMER = 167;
+	private static final int WINDOW_ID_SALES_ORDER = 143;
+
+	/**
+	 * https://idempiere.atlassian.net/browse/IDEMPIERE-4836
+	 * Unit testing for Document access test as per DocumentStatusAccess
+	 */
+	@Test
+	public void testDocumentStatusVisibility() {
+
+		/* Document Status with specific role and user access */
+		MDocumentStatus invoiceDS = new MDocumentStatus(Env.getCtx(), 0, getTrxName());
+		invoiceDS.setName("Invoice Document Status Unit Test");
+		invoiceDS.setEntityType(MEntityType.ENTITYTYPE_Dictionary);
+		invoiceDS.setAD_Table_ID(MInvoice.Table_ID);
+		invoiceDS.setAD_Window_ID(WINDOW_ID_INVOICE_CUSTOMER);
+		invoiceDS.setSeqNo(10);
+		invoiceDS.saveEx();
+
+		X_PA_DocumentStatusAccess invoiceDSAccess = new X_PA_DocumentStatusAccess(Env.getCtx(),0, getTrxName());
+		invoiceDSAccess.set_ValueOfColumn("PA_DocumentStatus_ID", invoiceDS.get_ID());
+		invoiceDSAccess.set_ValueOfColumn("AD_User_ID", USER_GARDENADMIN);
+		invoiceDSAccess.set_ValueOfColumn("AD_Role_ID", ROLE_GARDENWORLDADMIN);
+		invoiceDSAccess.saveEx();
+
+
+		/* Document Status with no user or role access, visible to all */
+		MDocumentStatus noticeDS = new MDocumentStatus(Env.getCtx(), 0,	getTrxName());
+		noticeDS.setName("order Document Status Unit Test");
+		noticeDS.setEntityType(MEntityType.ENTITYTYPE_Dictionary);
+		noticeDS.setAD_Table_ID(MOrder.Table_ID);
+		noticeDS.setAD_Window_ID(WINDOW_ID_SALES_ORDER);
+		noticeDS.setSeqNo(20);
+		noticeDS.saveEx();
+
+		/* Document Status with specific user access */
+		MDocumentStatus requestDS = new MDocumentStatus(Env.getCtx(), 0, getTrxName());
+		requestDS.setName("Notice Document Status Unit Test");
+		requestDS.setEntityType(MEntityType.ENTITYTYPE_Dictionary);
+		requestDS.setAD_Table_ID(MInvoice.Table_ID);
+		requestDS.setAD_Window_ID(WINDOW_ID_INVOICE_CUSTOMER);
+		requestDS.setSeqNo(30);
+		requestDS.saveEx();
+
+		X_PA_DocumentStatusAccess requestDSAccess = new X_PA_DocumentStatusAccess(Env.getCtx(),0, getTrxName());
+		requestDSAccess.set_ValueOfColumn("PA_DocumentStatus_ID", requestDS.get_ID());
+		requestDSAccess.set_ValueOfColumn("AD_Role_ID", ROLE_GARDENWORLDUSER);
```

Comment:
this should be changed to use the generated setter

Reply:
Fixed